cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -331,7 +323,13 @@ func getSchedulerDeployment(cfg *kubeadmapi.MasterConfiguration) ext.Deployment
 			},
 		},
 	}
-	return d
+}
+
+func prependSelfHostedFlock(name string, args []string) []string {
+	// TODO: flock has to be inside the container, and a shell must exist with this approach
+	// Since the flock binary is present at /bin/flock in a busybox image and /usr/bin/flock in a debian image, we can't hardcode the path
+	command := []string{"flock", "--exclusive", "--timeout=30", fmt.Sprintf("/var/lock/%s.lock", name)}
```

Comment:
I don't know how is a lock inside the container supposed to work and I cannot see `/var/lock` being mounted from the host or anything... If it is mounted from the host, please point me at it.